### PR TITLE
HHH-12375

### PR DESCRIPTION
**Pull request by dreab8**
https://hibernate.atlassian.net/browse/HHH-12375

@beikov what do you think?

### CODE REVIEW

**Comment by beikov** [comment]
Looks good and cleaner. Removing the path will have the same effect as putting null into the map so this should be fine.

**Comment by dreab8** [comment]
@sebersole what do u think, may it work or do you see problems with this approach?

File: hibernate-core/src/main/java/org/hibernate/persister/entity/AbstractPropertyMapping.java [hunk 1]
```diff
@@ -127,6 +131,18 @@ private void logDuplicateRegistration(String path, Type existingType, Type type)
 		}
 	}
 
+	private void logIncompatibleRegistration(String path, Type existingType, Type type) {
+		if ( LOG.isTraceEnabled() ) {
+			LOG.tracev(
+					"Skipped adding same named type incompatible property to base type [{0}] for property [{1}], existing type = [{2}], incoming type = [{3}]",
```
Comment (sebersole) [inline hibernate-core/src/main/java/org/hibernate/persister/entity/AbstractPropertyMapping.java]: Can we say something like this, instead?

````
"Skipped adding attribute [{1}] to base-type [{0}] as more than one sub-type defined the attribute using incompatible types (strictly speaking the attributes are not inherited); existing type = [{2}], incoming type = [{3}]"
````

Comment (dreab8) [reply hibernate-core/src/main/java/org/hibernate/persister/entity/AbstractPropertyMapping.java]: Thanks @sebersole,
I'll change it.

**Comment by cnsgithub** [comment]
@dreab8 I just tested your modifications to the original bugfix and can confirm that things are still working. Now it's time to release 5.2.16. :wink:

**Comment by dreab8** [comment]
Thanks @cnsgithub,

I'm going to do the release tomorrow.

**Comment by dreab8** [comment]
applied upstream
Thanks @sebersole , @beikov @cnsgithub

**Pull request closed without merging**